File: src/main/java/chessEngine/chess/piece/constantMovesPiece/king/BlackKing.java
package chessEngine.chess.piece.constantMovesPiece.king;

import chessEngine.chess.EnginePosition;
import chessEngine.chess.engineMove.field.Field;
import chessEngine.chess.piece.PieceColor;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public final class BlackKing extends King {
    public BlackKing(Field field, EnginePosition pos) {
        super(PieceColor.BLACK, pos, "bK", field);
        if (!correctFieldCoordinates(field.getHeight(), field.getWidth())) {
            throw new IllegalArgumentException("illegal height or width in constructor: " + field);
        }
    }
}

File: angular/src/app/chess/pieces/piece/king/blackKing/black-king.ts
import { Board } from "src/app/chess/board/board";
import { Field } from "src/app/chess/field/field";
import { PieceColor } from "../../../piece";
import { King } from "../king";

export class BlackKing extends King {

    iconPath: string = "assets/black_king.svg"
    color: PieceColor = PieceColor.Black;

    override moveTo(height: number, width: number) {
        var castlingMove = false
        if (this.fieldHeight == 0 && this.fieldWidth == 4 && height == 0) {
            if (width == 6) {
                this.moveRook(0, 7, 0, 5)
                castlingMove = true
            }
            else if (width == 2) {
                this.moveRook(0, 0, 0, 3)
                castlingMove = true
            }
        }
        
        this.fieldHeight = height;
        this.fieldWidth = width;

        if (!this.alreadyMoved) {
            this.alreadyMoved = true;
        }
        return castlingMove
    }


    constructor(fieldHeight: number, fieldWidth: number, board: Board) {
        super(fieldHeight, fieldWidth, board)
    }

}
